src/compiler/checker.ts

```diff
@@ -6870,17 +6874,6 @@ namespace ts {
             return type;
         }
 
-        /**
-         * Get type from reference to named type that cannot be generic (enum or type parameter)
-         */
-        function getTypeFromNonGenericTypeReference(node: TypeReferenceType, symbol: Symbol): Type {

```

Comment:
Why inline this function?

Reply:
Previously we would unconditionally enter this function. Now we will first **try**GetDeclaredTypeOfSymbol, and only continue into this section if the result is defined. So the only part of this function left is:
```ts
if (typeArguments) {
    error(node, Diagnostics.Type_0_is_not_generic, symbolToString(symbol));
    return unknownType;
}
```
which doesn't seem worthy of its own function.